File: Data Structures & Algorithms/validate-parentheses/submission-1.cs
public class Solution {
    public bool IsValid(string s) {
       if (s.Length == 0) return false;
        var stack = new Stack<char>();
        foreach (var ch in s)
        {
            if (ch == '(' || ch == '[' || ch == '{')
            {
                stack.Push(ch);
            }

            else
            {
                if (stack.Count != 0)
                {
                    char check = stack.Pop();
                    if (check == '(' && ch == ')' || check == '[' && ch == ']' || check == '{' && ch == '}')
                    {
                        continue;
                    }

                    else return false;
                }
                else return false;
            }
        }
        return stack.Count == 0;
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-1.ts
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures: number[]): number[] {
        const n = temperatures.length;
        let answer: number[] = new Array(n).fill(0);
        let stack: number[] = [];

        for (let i = 0; i < n; i++) {
            while (
                stack.length > 0 &&
                temperatures[i]! > temperatures[stack[stack.length - 1]!]!
            ) {
                let prevIdx = stack.pop()!;
                answer[prevIdx] = i - prevIdx;
            }
            stack.push(i);
        }

        return answer;
    }
}
